### PR TITLE
Colortool: add support for truecolor fg/bg and cursor

### LINKED ISSUE #1160: ColorTool does not set the default background color
# Environment

Microsoft Windows [Version 10.0.18362.145]
Windows Terminal built off master today, with ColorTool built off master today

I've built them both using release, Windows Terminal is deployed (Windows Terminal (Dev Build) launched from Start Menu).

# Steps to reproduce

ColorTool.exe -c
ColorTool.exe {any iterm scheme}

bug_report_1
bug_report_2

# Expected behavior

Color scheme should be set in local Windows Terminal instance.

# Actual behavior

Its not getting set.

Note that I tried this out in a regular PowerShell instance launched from Start Menu and it seems to do what it's supposed to (at least from what I can tell), in that it sets the color scheme. I noted that the background color doesn't appear to be correct, which when I searched for ColorTool bug reports, I found #1150 which seems to be consistent with the behavior that I was seeing.

### PR DESCRIPTION
**Pull request by LuanVSO**
Colortool: add support for truecolor fg/bg and cursor

## PR Checklist
* [x] Closes #1160
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
- add support to RGB values in FOREGROUND/BACKGROUND keys of ini colorschemes.
- add support to parse and change cursor color
- add support for true rgb fg/bg to colortool, will only work for vt and default console as there's no way to set rgb values through the win api.
- this pr does not add support for exporting an colorscheme with rgb fg/bg
- this is my first time using C# so any feedback is appreciated.

## Validation Steps Performed
manual testing

### CODE REVIEW

**Comment by LuanVSO** [comment]
is there any example of a Json color scheme?

**Comment by zadjii-msft** [comment]
> is there any example of a Json color scheme?

I'd look here: https://github.com/lukesampson/concfg

**Comment by LuanVSO** [comment]
> > is there any example of a Json color scheme?
> 
> I'd look here: https://github.com/lukesampson/concfg

do they even support rgb for fg/bg?

**Comment by zadjii-msft** [comment]
concfg likely doesn't include fg/bg at all, considering it was written for the vintage console which didn't originally support them _at all_ 😉

**Comment by LuanVSO** [comment]
should i "extend" the format to support it?

**Review by zadjii-msft — APPROVED** [review]

**Comment by zadjii-msft** [comment]
Looks great, thanks!

**Review by miniksa — CHANGES_REQUESTED** [review]
Just the one question about the set to -1 versus delete

File: src/tools/ColorTool/ColorTool/ConsoleTargets/DefaultConsoleTarget.cs [hunk 1]
```diff
@@ -21,11 +21,39 @@ public void ApplyColorScheme(ColorScheme colorScheme, bool quietMode)
...
+
                 consoleKey.SetValue("ScreenColors", screenColors, RegistryValueKind.DWord);
+
+                if (colorScheme.ColorTable[fgidx] != colorScheme.ConsoleAttributes.Foreground.Value)
+                {
+                    consoleKey.SetValue("DefaultForeground", colorScheme.ConsoleAttributes.Foreground.Value, RegistryValueKind.DWord);
+                }
+                else
+                {
+                    consoleKey.SetValue("DefaultForeground", -1, RegistryValueKind.DWord);
```
[the remark this answers is not in the document]
Comment (LuanVSO) [reply src/tools/ColorTool/ColorTool/ConsoleTargets/DefaultConsoleTarget.cs:45]: i was replicating the behaviour when you disable it from the conhost property page.
but i will change it to delete the key if you prefer this

[the remark this answers is not in the document]
Comment (miniksa) [reply src/tools/ColorTool/ColorTool/ConsoleTargets/DefaultConsoleTarget.cs:45]: Oh. Well. Nevermind me then. Let's roll with it.

**Review by miniksa — APPROVED** [review]

**Comment by itsmeboy0** [comment]
When will we expect this change to be released?

**Pull request merged**